experiments/experiments.go

```diff
@@ -34,6 +36,11 @@ var (
 	experiments = make(map[string]bool, len(Available))
 )
 
+func WithUndo(key string) func() {
+	Enable(key)
+	return func() { Disable(key) }
+}
+
```

Comment:
I also think it's a bit evil to enable the experiment as a side effect and not mention it in the method name. Then when you're reading the code 
```go
defer experiments.WithUndo(experiments.GitMirrors)()
```
It reads like you're deferring the experiment ("with undo", whatever that is).

Maybe this is more clear?
```suggestion
func EnableReturningUndo(key string) func() {
	Enable(k)
	return func() { Disable(key) }
}
```

Then I would use it like this:
```go
undo := experiments.EnableReturningUndo(experiments.GitMirrors)
defer undo()
```